### PR TITLE
Moving advanced constitutive laws from StructuralMechanics to ConstitutiveLawsApp

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
This PR finally moves the advanced constitutive laws from the StructuralMechanicsApplication to the ConstitutiveLawsApplication
This was discussed for a long time

Follow up PR of #7404

Of course in the future more cleanups will be necessary but this is a start :)

For the user nothing changes, StructuralMechanics imports the ConstitutiveLawsApp if available

### CODE REVIEW

**Comment by philbucher** [comment]
one more comment:
I would merge this asap as it has a high risk for conflicts

File: applications/ConstitutiveLawsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp:886]: can we add const here?

**Review by loumalouomega — CHANGES_REQUESTED** [review]
IMO test should be moved to the new app

File: applications/ConstitutiveLawsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp
Comment (philbucher) [reply applications/ConstitutiveLawsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp:886]: probably but this should be a different PR

**Comment by philbucher** [comment]
> IMO test should be moved to the new app

Sure, but this will require some refactoring, hence I would do it in a future PR, this one is already pretty large
As I wrote this is only a first step

Also note that the CI still runs all the tests, nothing has changed in that regard

**Review by loumalouomega — DISMISSED** [review]

**Comment by marandra** [comment]
We should wait for PR changes to the advanced CLs until this is merged, right?

**Comment by philbucher** [comment]
> We should wait for PR changes to the advanced CLs until this is merged, right?

yes, thats why I would merge this asap

**Comment by philbucher** [comment]
any comment @AlejandroCornejo @RiccardoRossi ?
If not I would go ahead to avoid conflicts

**Review by tteschemacher — APPROVED** [review]
Really nice. Thanks a lot. That will help a lot to shape the structure of our beloved KRATOS.

I would like to ask to merge as quick as possible. There's always some slight movements in the CL and this would create insane dependencies and would make it really difficult to keep this PR up to date...

Also, I'm currently preparing some PRs and would of course prefer to put those directly, here.

**Comment by AlejandroCornejo** [comment]
HI! Wow thank you for the work done! I'm currently implementing a new CL but I suppose that I will be able to do merge and moving all the stuff... @RiccardoRossi

**Pull request merged**
